src/order/minimal.lean

```diff
@@ -117,6 +118,38 @@ begin
   rwa of_not_not (λ hab, ht ha (h hb) hab hr),
 end
 
+lemma exists_mem_maximals (hr₁ : reflexive r) (hr₂ : transitive r) (hr₃ : anti_symmetric r)
+  {s : set α} (ih : ∀ c ⊆ s, is_chain r c → c.nonempty → ∃ ub ∈ s, ∀ z ∈ c, r z ub) (x ∈ s) :
+  ∃ x' ∈ maximals r s, r x x' :=
```

Comment:
This is almost the same as [zorn_nonempty_partial_order₀](https://leanprover-community.github.io/mathlib_docs/order/zorn.html#zorn_nonempty_partial_order%E2%82%80); you can make a `partial_order` instance inside the proof and use it.

Reply:
My plan is to go the other way, actually. I want to restate Zorn's lemma using `minimals`.

Reply:
Here's the PR to redefine `minimals` and `maximals` as was discussed on Zulip: #16103.